src/mlpack/methods/linear_svm/linear_svm_main.cpp

```diff
@@ -0,0 +1,417 @@
+
+  // Tolerance needs to be positive.
+  RequireParamValue<double>("tolerance", [](double x) { return x >= 0.0; },
+      true, "tolerance must be positive or zero");
+
+  // Optimizer has to be L-BFGS or ParallelSGD.
+  #ifdef HAS_OPENMP
+  RequireParamInSet<string>("optimizer", { "lbfgs", "psgd" },
+      true, "unknown optimizer");
+
+  if (optimizerType != "psgd")
+  {
+    if (CLI::HasParam("step_size"))
+    {
+      Log::Warn << PRINT_PARAM_STRING("step_size") << " ignored because "
+          << "optimizer type is not 'psgd'." << std::endl;
+    }
+    if (CLI::HasParam("shuffle"))
+    {
+      Log::Warn << PRINT_PARAM_STRING("shuffle") << " ignored because "
+          << "optimizer type is not 'psgd'." << std::endl;
+    }
+  }
+
+  // Step Size must be positive.
+  RequireParamValue<double>("step_size", [](double x) { return x >= 0.0; },
+      true, "step size must be positive");
+
+  #endif
+
+  // Lambda must be positive.
+  RequireParamValue<double>("lambda", [](double x) { return x >= 0.0; },
+      true, "lambda must be positive or zero");
+
+  // Number of Classes must be Non-Negative
+  RequireParamValue<int>("number_of_classes", [](int x) { return x >= 0; },
+                         true, "number of classes must be greater than or "
+                         "equal to 0 (equal to 0 in case of unspecified.)");
+
+  // Delta must be positive.
+  RequireParamValue<double>("delta", [](double x) { return x >= 0.0; }, true,
+      "Margin of difference between correct class and other classes");
+
+  // These are the matrices we might use.
+  arma::mat trainingSet;
+  arma::Row<size_t> labels;
+  arma::mat testSet;
+  arma::Row<size_t> predictions;
+
+  // Load data matrix.
+  if (CLI::HasParam("training"))
+    trainingSet = std::move(CLI::GetParam<arma::mat>("training"));
+
+  // Check if the labels are in a separate file.
+  if (CLI::HasParam("training") && CLI::HasParam("labels"))
+  {
+    labels = std::move(CLI::GetParam<arma::Row<size_t>>("labels"));
+    if (trainingSet.n_cols != labels.n_cols)
+    {
+      Log::Fatal << "The labels must have the same number of points as the "
+          << "training dataset." << endl;
+    }
+  }
+  else if (CLI::HasParam("training"))
+  {
+    // Checking the size of training data if no labels are passed.
+    if (trainingSet.n_rows < 2)
+    {
+      Log::Fatal << "Can't get labels from training data since it has less "
+          << "than 2 rows." << endl;
+    }
+
+    // The initial predictors for y, Nx1.
+    labels = arma::conv_to<arma::Row<size_t>>::from(
+        trainingSet.row(trainingSet.n_rows - 1));
+    trainingSet.shed_row(trainingSet.n_rows - 1);
+  }
+
+  const size_t numClasses = NumberOfClasses(
+       (size_t) CLI::GetParam<int>("number_of_classes"), labels);
+
+  // Load the model, if necessary.
+  LinearSVM<>* model;
+  if (CLI::HasParam("input_model"))
+    model = CLI::GetParam<LinearSVM<>*>("input_model");
+  else
+  {
+    model = new LinearSVM<>;
+  }
```

Comment:
If we use braces for the `else`, we should also use braces for the `if` (or, we should use braces for neither).  Sorry for the picky style comment. :)